From 66c6e70316cc5604a682bc6273d7db6375d22650 Mon Sep 17 00:00:00 2001
From: Jason Andryuk <jandryuk@gmail.com>
Date: Fri, 20 Sep 2019 09:55:13 -0400
Subject: [PATCH 11/26] oxt: flask: Add dm_delegate_device interface

Use of PCI passthrough with a stubdom required dom0 to delegate to the
stubdom, which in turn delegates to the actual domain.  Provide a helper
interface.

Signed-off-by: Jason Andryuk <jandryuk@gmail.com>
---
 tools/flask/policy/modules/xen.if | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/tools/flask/policy/modules/xen.if b/tools/flask/policy/modules/xen.if
index fe20ad4b14..1116fdaeea 100644
--- a/tools/flask/policy/modules/xen.if
+++ b/tools/flask/policy/modules/xen.if
@@ -239,6 +239,13 @@ define(`delegate_devices', `
     allow $1 $2:resource { add remove };
 ')
 
+# dm_delegate_devices(priv-domain, dm-domain, target-domain)
+#   Allow devices to be delegated through a stubdom
+define(`dm_delegate_devices', `
+    delegate_devices($1, $2)
+    delegate_devices($2, $3_target)
+')
+
 # argo_comms(domA, domB)
 #   Allow two domains to talk argo to each other
 define(`argo_comms', `
-- 
2.21.0

